uint32 num_points
float32[] px
float32[] py
---
float32[] angles
